### DISCUSSION TITLE
Facing issues running the simulator

### DISCUSSION THREAD

**Discussion by SaffatZabin-17** [ChampSim/ChampSim#501] [Q&A] [answered]
Hi. I am facing issues running the simulator. I have downloaded the vcpkg dependencies and compiled using the config file in the repository. I have downloaded a DPC-3 Trace file, specifically the [603.bwaves_s-1080B.champsimtrace.xz](https://dpc3.compas.cs.stonybrook.edu/champsim-traces/speccpu/603.bwaves_s-1080B.champsimtrace.xz) trace file. 

But when I try to execute the binary using the bin/champsim command as instructed, it says:   `bash: $: command not found`. 

Point to be noted: 

1. The github repo of the simulator is not cloned in the root directory, rather in another drive. 
2. I have tried putting the trace file in the home directory of the simulator, and in the bin folder. 
3. I have tried a fresh compilation of the simulator multiple times. 
4. The config.json file provided in the repository was used to compile it. No modification was made. 
5. My linux version is: Debian 11 (Bullseye)

TIA

**Comment by maccoymerrell** [comment, marked as answer]
A couple of questions. 
1. Is there anything in the bin directory at all? That is where the champsim binary would be.
2. Did you run make? The exact order should be git clone -> git submodule init -> git submodule update -> ./vcpkg/bootstrap -> ./vcpkg/vcpkg install -> ./config.sh config.json -> make -> run
3. Did you try to run ./bin/champsim instead of bin/champsim (if champsim is in the bin directory)?

**Reply by SaffatZabin-17** [reply]
1. Yes, the champsim binary is in the bin folder. 
2. Yes. I did run make. I followed the order mentioned in the readme.md file
3. Yes, I did. It did not work

**Reply by maccoymerrell** [reply]
Maybe run chmod +x bin/champsim
Then ./bin/champsim

**Reply by maccoymerrell** [reply]
If not, could you send the exact command you are running that gives you the error?

**Reply by SaffatZabin-17** [reply]
I did as you suggested, but it did not work. The commands I have used are below in order:

`git submodule update --init`

`vcpkg/bootstrap-vcpkg.sh`

`vcpkg/vcpkg install`

`./config.sh champsim_config.json` (The default config file in the repository)

`make`

`chmod +x bin/champsim`

`./bin/champsim --warmup_instructions 200000000 --simulation_instruction 500000000 603.bwaves_s-1080B.champsimtrace.xz`   (The trace file is in the ChampSim home directory. I am running these commands from the same directory)

**Reply by maccoymerrell** [reply]
Are you sure these are the exact commands you are executing?

Based on your bash error, are you sure you are not trying to run
$ ./bin/champsim --warmup_instructions 200000000 --simulation_instruction 500000000 603.bwaves_s-1080B.champsimtrace.xz
instead of 
./bin/champsim --warmup_instructions 200000000 --simulation_instruction 500000000 603.bwaves_s-1080B.champsimtrace.xz

The $ represents the beginning of the command line in bash. Its not part of a command.

**Reply by SaffatZabin-17** [reply]
Ah, I apologize for this. It was an really bad oversight on my part. I have run the previous commands by typing manually. However, as the last one was quite large, I copied it and pasted it in my terminal :)  Sorry for that. Should've paid more attention. 

Thank you very much for your time and effort.
